qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationElwinTab.cpp

```diff
@@ -194,35 +114,55 @@ void InelasticDataManipulationElwinTab::runComplete(bool error) {
   if (!error) {
     if (!m_view->isGroupInput()) {
       m_model->ungroupAlgorithm("IDA_Elwin_Input");
+    } else {
+      std::string outputNames =
+          checkForELTWorkspace()
+              ? m_model->getOutputWorkspaceNames()
+              : m_model->getOutputWorkspaceNames().substr(0, m_model->getOutputWorkspaceNames().find_last_of(','));
+      m_model->groupAlgorithm(outputNames, "IDA_Elwin_Output");
     }
 
     setOutputPlotOptionsWorkspaces(getOutputWorkspaceNames());
 
-    if (m_view->getNormalise())
-      checkForELTWorkspace();
-
+    if (m_view->getNormalise() && !checkForELTWorkspace())
+      m_view->showMessageBox("ElasticWindowMultiple successful. \nThe _elt workspace "
+                             "was not produced - temperatures were not found.");
   } else {
     m_view->setSaveResultEnabled(false);
   }
 }
 
-void InelasticDataManipulationElwinTab::checkForELTWorkspace() {
+bool InelasticDataManipulationElwinTab::checkForELTWorkspace() {
   auto const workspaceName = getOutputBasename() + "_elt";
-  if (!WorkspaceUtils::doesExistInADS(workspaceName))
-    m_view->showMessageBox("ElasticWindowMultiple successful. \nThe _elt workspace "
-                           "was not produced - temperatures were not found.");
+  if (!WorkspaceUtils::doesExistInADS(workspaceName)) {
+    return false;
+  }
+  return true;
+}
+
+std::string InelasticDataManipulationElwinTab::prepareOutputPrefix(std::vector<std::string> workspaceNames) {
+  std::transform(workspaceNames.cbegin(), workspaceNames.cend(), workspaceNames.begin(),
+                 [](auto &name) { return name.substr(0, name.find_first_of('_')); });
+  std::regex re("\\d+");
```

Comment:
I think this regex can be treated as a constant and moved to the top of the file in an annonymous namespace? Instantiating these repeatedly can sometimes be expensive